File: app/csharp/AccountingSystem.Infrastructure/Entities/TaxTransaction.cs
namespace AccountingSystem.Infrastructure.Entities;

/// <summary>
/// 課税取引エンティティクラス
/// データベースの日本語カラム名と英語プロパティ名をマッピング
/// </summary>
public record TaxTransaction
{
    /// <summary>
    /// 課税取引コード（01:課税、02:非課税、03:免税、04:不課税）
    /// </summary>
    public string TaxCode { get; init; } = string.Empty;

    /// <summary>
    /// 課税取引名
    /// </summary>
    public string TaxName { get; init; } = string.Empty;

    /// <summary>
    /// 税率（0.10 = 10%）
    /// </summary>
    public decimal TaxRate { get; init; }

    /// <summary>
    /// 説明
    /// </summary>
    public string? Description { get; init; }

    /// <summary>
    /// 有効フラグ
    /// </summary>
    public bool IsActive { get; init; } = true;

    /// <summary>
    /// 作成日時
    /// </summary>
    public DateTime CreatedAt { get; init; }

    /// <summary>
    /// 更新日時
    /// </summary>
    public DateTime UpdatedAt { get; init; }
}

File: app/fsharp/AccountingSystem.Domain/Models/TaxTransaction.fs
namespace AccountingSystem.Domain.Models

/// <summary>
/// 課税取引エンティティ
/// </summary>
[<CLIMutable>]
type TaxTransaction = {
    TaxCode: string                    // 課税取引コード
    TaxName: string                    // 課税取引名
    TaxRate: decimal                   // 税率
    Description: string option         // 説明
}

/// TaxTransaction エンティティのファクトリ関数
module TaxTransaction =
    let create taxCode taxName taxRate =
        {
            TaxCode = taxCode
            TaxName = taxName
            TaxRate = taxRate
            Description = None
        }
